org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/LanguageServersTest.java

```diff
@@ -148,7 +148,7 @@ public synchronized CompletableFuture<Hover> hover(HoverParams position) {
 		List<String> hovers = result.join();
 		
 		assertTrue(hovers.contains("HoverContent1"));
-		assertFalse(hovers.contains("HoverContent2"));
+		assertFalse(hovers.contains(null));
```

Comment:
why do we need to change "HoverContents2" with `null`? I do not understand this part.

Reply:
I put it as part of the first commit (which got rid of the superfluous call to shutdown the dispatch threads) as it's maintenance not really part of this main change. I realised the test is wrong, or rather incomplete - `testCollectAllExcludesNulls()` should be making sure that any null response from a language server gets filtered out of the returned list. The test (owing to some cut-and-paste errors and a brain fade on my part) wasn't ensuring that properly. If you look at the mock it can never generate 'HoverContent2' it is supposed to be generating null on the second call, and the test should explicitly be testing for the null getting filtered out.

I just realised this in the course of the main change - I had made some errors in earlier drafts of this work ensuring that the null semantics were preserved when code was rewritten to the new API, so I checked this carefully and spotted the flawed test.

Can split this fix out if you think it is still unclear - I kept it as a separate commit to emphasise it was an unrelated change.